custom_components/ha_hatch/sensor.py

```diff
@@ -39,3 +40,28 @@ def _update_local_state(self):
         _LOGGER.debug(f"updating state:{self.rest_device}")
         self._attr_native_value = self.rest_device.battery_level
         self.async_write_ha_state()
+
+class HatchCharging(RestEntity, SensorEntity):
+    _attr_icon = "mdi:power-plug"
+
+    def __init__(self, rest_device: RestPlus):
+        super().__init__(rest_device, "Charging Status")
+
+    def _update_local_state(self):
+        if self.platform is None:
+            return
+        _LOGGER.debug(f"updating state:{self.rest_device}")
+        if self.rest_device.charging_status is 0:
```

Comment:
an example of using enum to keep these codes in the library is https://github.com/dahlb/hatch_rest_api/blob/master/src/hatch_rest_api/const.py#L11 RestMiniAudioTrack if you want to push this logic into the library is should be very easy basing the change off that example but doesn't have to be in this pr

Reply:
Okay I can look into enum and incorporate it into a future PR.